_inc/lib/class.jetpack-user-event-tracking.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+class Jetpack_User_Event_Tracking {
+
+	private static $_cache = array();
+	private static $KEY = 'jetpack_event_tracking';
+
+	static function is_enabled( $user_id ) {
+		if ( isset( self::$_cache[ $user_id ] ) ) {
+			return self::$_cache[ $user_id ];
+		}
+		$user_tracking = get_user_meta( $user_id, self::$KEY, true );
+		if ( ! is_numeric( $user_tracking ) ) {
+			$user_tracking = self::default_value();
+		}
+		self::$_cache[ $user_id ] = (bool) $user_tracking;
+		return (bool) $user_tracking;
+	}
+
+	static function is_disabled( $user_id ) {
+		return ! self::is_enabled( $user_id );
+	}
+
+	static function disable( $user_id ) {
+		// user opted out
+		return self::set( $user_id, 0 );
+	}
+
+	static function enable( $user_id ) {
+		// user opted in
+		return self::set( $user_id, 1 );
+	}
+
+	static private function set( $user_id, $value ) {
+		self::$_cache[ $user_id ] = (bool) $value;
+		return update_user_meta( $user_id, self::$KEY, $value );
```

Comment:
Why are you casting this as boolean when you save it to the cache, but not when you update the user meta data?

Reply:
Because I want to be able to distinguish between a set value (0,1) vs the default value (false)

Reply:
I see, thanks!